src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs

```diff
@@ -232,15 +232,25 @@ public void LoadTables(byte[] tableBytes, HuffmanScanDecoder huffmanScanDecoder)
             using var stream = new BufferedReadStream(this.Configuration, ms);
 
             // Check for the Start Of Image marker.
-            stream.Read(this.markerBuffer, 0, 2);
+            int bytesRead = stream.Read(this.markerBuffer, 0, 2);
+            if (bytesRead != 2)
+            {
+                JpegThrowHelper.ThrowInvalidImageContentException("Not enough data to read the SOI marker");
+            }
```

Comment:
I think we can remove this and similar checks from other marker parsing methods as it's now guaranteed that stream has at least `remaining` bytes available.

Reply:
Ah... That method is actually called from TIFF only.

Reply:
But we still can transform 2 read checks to 1 `AvailableBytes` check :)
I can't comment on non diff lines but that method creates stream:

```cs
using var ms = new MemoryStream(tableBytes);
```

and then do 2 reads of 2 bytes with sequential if checks.

We can do this check:

```cs
if(tableBytes.Length < 4)  throw ...
```

before even creating a stream and remove 2 checks afters reads as those would always be true.

Reply:
Cool. Will have a look.